### PR TITLE
Update Python version supported

### PR DESCRIPTION
**Pull request by lmazuel**

### CODE REVIEW

**Comment by Petermarcu** [comment]
@annatisch do you mind doing a quick review of this so we can get it merged?

File: docs/python/_includes/example_client.py [hunk 1]
```diff
@@ -28,8 +27,7 @@ def __init__(self, name, size):
         self.size = size
 
     @classmethod
-    def from_response(self, response):
-        # type: (azure.core.pipeline.HttpResponse) -> Thing
+    def from_response(self, response: "azure.core.pipeline.HttpResponse") -> "Thing":
```
Comment (annatisch) [inline docs/python/_includes/example_client.py:30]:
```suggestion
    def from_response(self, response: "azure.core.rest.HttpResponse") -> "Thing":
```

File: docs/python/_includes/example_client.py [hunk 2]
```diff
@@ -39,20 +37,19 @@ def __repr__(self):
         # truncate the output at 1024 characters.
         return json.dumps(self.__dict__)[:1024]
 
-class ExampleClient(object):
+class ExampleClient:
 
-    def __init__(self, endpoint, credential, **kwargs):
-        # type: (str, azure.core.credential.TokenCredential, **Any) -> None
+    def __init__(self, endpoint: str, credential: "azure.core.credential.TokenCredential", **kwargs) -> None:
```
Comment (annatisch) [inline docs/python/_includes/example_client.py:42]:
```suggestion
    def __init__(self, endpoint: str, credential: "azure.core.credentials.TokenCredential", **kwargs) -> None:
```

File: docs/python/_includes/example_client.py [hunk 3]
```diff
@@ -39,20 +37,19 @@ def __repr__(self):
...
         :param endpoint: Endpoint to connect to.
         :type endpoint str:
         :param credential: Credentials to use when connecting to the service.
         :type credential: ~azure.core.credentials.TokenCredential
         :keyword apiversion: API version to use when talking to the service. Default is '2020-12-31'
-        :type apiversion: str
+        :paramtype apiversion: str
         :keyword transport: HttpTransport to use. Default is ~transports.RequestsHttpTransport.
-        :type transport: ~transports.HttpTransport
+        :paramtype transport: ~transports.HttpTransport
```
Comment (annatisch) [inline docs/python/_includes/example_client.py:52]:
```suggestion
        :paramtype transport: ~azure.core.pipeline.transport.HttpTransport
```

File: docs/python/_includes/example_client.py [hunk 4]
```diff
@@ -39,20 +37,19 @@ def __repr__(self):
...
-    def __init__(self, endpoint, credential, **kwargs):
-        # type: (str, azure.core.credential.TokenCredential, **Any) -> None
+    def __init__(self, endpoint: str, credential: "azure.core.credential.TokenCredential", **kwargs) -> None:
         """Create a new example client instance
 
         :param endpoint: Endpoint to connect to.
         :type endpoint str:
         :param credential: Credentials to use when connecting to the service.
         :type credential: ~azure.core.credentials.TokenCredential
         :keyword apiversion: API version to use when talking to the service. Default is '2020-12-31'
```
Comment (annatisch) [inline docs/python/_includes/example_client.py:49]:
```suggestion
        :keyword api_version: API version to use when talking to the service. Default is '2020-12-31'
```

File: docs/python/_includes/example_client.py [hunk 5]
```diff
@@ -39,20 +37,19 @@ def __repr__(self):
...
-        :type apiversion: str
+        :paramtype apiversion: str
```
Comment (annatisch) [inline docs/python/_includes/example_client.py:50]:
```suggestion
        :paramtype api_version: str
```

File: docs/python/_includes/example_client.py [hunk 6]
```diff
@@ -91,17 +86,15 @@ def get_thing(self, name, **kwargs):
         response = self._pipeline.send(request, **kwargs) 
         return model_factory(response)
 
-    def list_things(self, **kwargs):
-        # type: (**Any) -> azure.core.ItemPaged[Thing]
+    def list_things(self, **kwargs) -> "azure.core.ItemPaged[Thing]":
```
Comment (annatisch) [inline docs/python/_includes/example_client.py:89]:
```suggestion
    def list_things(self, **kwargs) -> "azure.core.paging.ItemPaged[Thing]":
```

File: docs/python/design.md [hunk 7]
```diff
@@ -519,13 +515,11 @@ class ConfigurationSetting(object):
     :vartype value: object
     """
 
-    def __init__(self, name, value):
-        # type: (str, object) -> None
+    def __init__(self, name: str, value: Any):
```
Comment (annatisch) [inline docs/python/design.md:518]:
```suggestion
    def __init__(self, name: str, value: object):
```

**Review by annatisch — APPROVED** [review]

**Comment by lmazuel** [comment]
Merging , as the CI issues are on links on pages not touched by this PR

**Pull request merged**